File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        // negative number is not a polindrome
        if (x < 0) {
            return false;
        }

        long reversed = 0;
        int xcopy = x;
        
        // reverse x 
        while (x > 0) {
            reversed = reversed * 10 + x % 10;
           x = x / 10; 
        }

        return xcopy == reversed;
    }
};

File: 0009-palindrome-number/0009-palindrome-number.py
class Solution(object):
    def isPalindrome(self, x):
        x = str(x)
        if len(x) < 2:
            return True
        
        i = 0
        j = len(x) - 1

        # check the digits from edges to mid
        while i < j:
            if x[i] != x[j]:
                return False
            i += 1
            j -= 1

        return True
        